pgsanity/sqlprep.py

```diff
@@ -17,14 +17,14 @@ def prepare_sql(sql):
         # decide where we are
```

Comment:
As you alluded to in your description, I'm on the fence about including this.  On the one hand, psql is pervasive, but on the other hand, it's simply not valid SQL.  There have been a few other pull requests / feature requests in the past to add support for expanding the syntax checker to support psql-only features: #10 .

I suppose I would be open to including this with a non-default flag.  So, one would need to pass a flag on the command-line to do this conversion from psql meta-commands to comments.  Without the flag it would treat these as errors.

For what it's worth, I'm impressed that you were able to dissect this code that I wrote.  It's really not great and is fairly hard to follow.

While reading the psql docs about meta-commands it looks like there is support for having a double backslash stop treating the commands as psql and switch back to SQL.  Any interest in supporting that as well?

From http://www.postgresql.org/docs/9.2/static/app-psql.html:
"Parsing for arguments stops at the end of the line, or when another unquoted backslash is found. An unquoted backslash is taken as the beginning of a new meta-command. The special sequence \\ (two backslashes) marks the end of arguments and continues parsing SQL commands, if any. That way SQL and psql commands can be freely mixed on a line. But in any case, the arguments of a meta-command cannot continue beyond the end of the line."